sklearn/pipeline.py

```diff
@@ -622,9 +626,16 @@ def classes_(self):
         return self.steps[-1][-1].classes_
 
     def _more_tags(self):
-        # check if first estimator expects pairwise input
-        estimator_tags = self.steps[0][1]._get_tags()
-        return {'pairwise': estimator_tags.get('pairwise', False)}
+        # Check if first estimator expects pairwise input
+        transformer = self.steps[0][1]
+        # The transformers may be None
+        transformer_tags = transformer._get_tags() if transformer else {}
+
+        estimator = self.steps[-1][1]
+        estimator_tags = estimator._get_tags()
```

Comment:
Do we have a test for this logic?

Reply:
I have not seen this logic in the tests for the `pipeline` module.